### DISCUSSION TITLE
Parquet format for spectral library storage

### DISCUSSION THREAD

**Discussion by enonimos** [vdemichev/DiaNN#1184] [General]
Hi,

I have read from a previous question/post that the latest DIA-NN 1.9.1 starts to use *.parquet file for storing spectral libraries. This is so much more efficient so thank you for that feature.  I have been able to get the parquet file when using libraries in tsv format, however,  when generating DIANN predicted libraries from a FASTA file, it seems to only save it in speclib. Is this working as intended, or am I missing a step to be able to store in silico predicted libaries as parquet. 

Thank you for the continued development of DIA-NN.

Todd

**Comment by vdemichev** [comment]
Hi Todd,

Yes, this is by design. The reason is, .parquet is fantastic, however it is still taking some time to read & write. In contrast, just directly writing/reading C++ structures on disk (that's what .speclib is) is very quick and is hence better suitable for huge libraries, with the difference apparent when the lib is on a fast SSD. Since we want to support libraries with 100M+ precursors (next DIA-NN versions will have fold-change lower RAM consumption, allowing that on a regular desktop), these would be very slow to read from .parquet - but fine with .speclib. In contrast, DIA-based libs are necessarily limited to confidently detected precursors and are thus <1M in size, hence speed is not an issue.

Best,
Vadim

**Reply by enonimos** [reply]
Thank you Vadim, that makes sense.
Todd

**Comment by dkueltz** [comment]
Hi Vadim,

I am trying to import a DIA-NN generated spectral library into Skyline. I am using DIA-NN 1.9.2, which I believe is the latest version. However  the Skyline button in DIA-NN does not work (cannot find either Skyline nor Dialy). Also, the output library generated by DIA-NN is only in parquet format and there seems to be no option for generating it in speclib format, which would be readable by Skyline. I wonder how to change the parquet format to a speclib format or if there is any other way to import DIA-NN library results into Skyline.

Thanks for any suggestions,
Dietmar Kültz (UC Davis)

**Reply by vdemichev** [reply]
Hi Dietmar,

Please rather use the .skyline.speclib library which DIA-NN exports when loading any lib, including .parquet. As far as I know, Skyline does not support parquet libs at the moment.

Best,
Vadim

**Reply by vdemichev** [reply]
In general visualisation with Skyline will work just by pressing the 'Skyline' button (if MBR, proper FASTA, etc).

**Comment by dkueltz** [comment]
Hi Vadim,

Thanks for the feedback. For some reason, when I run only a single data file no speclib was generated but it worked when I loaded multiple data files. Perhaps I enetered something incorrectly before. At any rate, it works now and I can import the speclib into skyline without any problems.

Best,
Dietmar

**Comment by SStoychev01** [comment]
Hi Vadim,

Related to the above, When I try to build a spec library in Skyline using the .parquet.skyline.spec.lib I see this error "ERROR: file 30DaysExploris.parquet does not have a column called Global.Q.Value" Is there a work aroud?

Thanks,

Stoyan

**Reply by vdemichev** [reply]
Hi Stoyan,

I am not sure what's being done, but Skyline tries to find the main report from DIA-NN and then confuses it with a library .parquet file.
If the goal is to import the whole DIA-NN analysis, please use the 'Skyline' button in DIA-NN (and also MBR in DIA-NN), otherwise I would suggest to place skyline.speclib to a separate folder with no other DIA-NN output in there, maybe then works?

Best,
Vadim

**Reply by SStoychev01** [reply]
Thanks Vadim, the aim is to build a spectral library in Skyline using the the .parquet.skyline.speclib. I tried to have this file on it's own but Skyline requires the .parquet file also. 
![SkylineError](https://github.com/user-attachments/assets/a1a33150-aaf5-4378-a021-285979a12b5a)
